presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java

```diff
@@ -781,6 +798,42 @@ private SpecialCallResult tryHandleCast(CallExpression callExpression, List<Obje
                 return changed(null);
             }
 
+            // Simplify nested casts to varchar types.
+            if (optimizationLevel == OPTIMIZED && source instanceof CallExpression && isVarcharType(targetType)) {
```

Comment:
Why not also EVALUATED?  Though EVALUATED is intended for evaluating the expression into constant, so I can understand this optimization is not really necessary (cast will be evaluated anyway). But perhaps, add a comment here?

Reply:
We have three phases for optimizing an expression:
1. SERIALIZABLE, which happens only in coordinator when we create the query plan. At this stage, we want to make sure the changes is minimal enough to cause any type inconsistency/serialization problems, ... etc.
2. OPTIMIZED, this happens in worker before we compile the expression to bytecode. Here, assigning value to coercible type is allowed.
3. EVALUATED, this phase is used to extract constant values out of expression. The expression is never be intended to be used again in planning/byte code generation.

Reply:
Thank you for the explanation, I restricted it to OPTIMIZED based on https://github.com/prestodb/presto/pull/14408#issuecomment-621552456, I am not opposed to allowing EVALUATED in the condition.